test/unit_test/options_test.cc

```diff
@@ -83,3 +83,106 @@ BOOST_AUTO_TEST_CASE(create_argument_group) {
   BOOST_CHECK_EQUAL(ag.m_options[3]->m_keep, true);
   BOOST_CHECK_EQUAL(ag.m_options[3]->m_type_hash, typeid(decltype(loc2)).hash_code());
 }
+
+BOOST_AUTO_TEST_CASE(name_extraction_from_option_group)
+{
+  char loc;
+  std::vector<std::string> loc2;
+  option_group_definition ag("g1");
+  ag(make_option("im_necessary", loc).keep().necessary());
+  ag(make_option("opt2", loc));
+  ag.add(make_option("opt3", loc2));
+  ag.add(make_option("opt4", loc2).keep());
+
+  auto name_extractor = options_name_extractor();
+  // result should always be false
```

Comment:
Why should this be false even if there is a 'necessary' option added?

Reply:
options_name_extractor.add_parse_and_check_necessary will always return false since it's sole purpose is to extract the string name from an options_group_definition and is not to be confused with other implementations of options_i

Reply:
it could also throw asides from returning false